### ISSUE TITLE
How to connect to MessageReceived signal from C++?

### ISSUE THREAD

**Issue by TheSDTM** [Larpon/QtFirebase#71]
Hello. How can I receive message (and token) inside my application? For now I do like this:

`
auto firebase = QtFirebaseMessaging::instance();
Notification handler;
QObject::connect(firebase, &QtFirebaseMessaging::messageReceived, &handler, &Notification::messageReceived);
`

Where Notification is my class. But I do not receive any signal. Also, no notification when App is opened.

**Comment by isipisi281** [comment]
You want to connect from outside?

**Comment by TheSDTM** [comment]
Yes

**Comment by Larpon** [comment]
If you can connect via QML it should indeed get the same results when connecting in C++

**Comment by Larpon** [comment]
@adolby - I can see why you'd need to be able to send out the same message more than once in the same app run (although it might be user bombardment, if not for testing purposes). We could add a bool for the emit upon data received. Something like `alwaysEmitDataChanged` with a default to `false`?

It also makes sense to pass the token to the signal as a parameter - it's useful - also from QML

**Comment by adolby** [comment]
Yeah, the flag for always emitting would be helpful for me-- the use case is that the message is a command from a server to tell the app to synchronize data, which really doesn't need any more data then the command itself.

It's also used for sending data only and not in messages shown to the user-- I agree that that would probably be bad to continue showing the same notification message to users.

We could have just as easily added a timestamp to the message data to have avoided this issue, though, which might be a better way to have solved the problem.

**Comment by TLCoetzer** [comment]
When not running from QML but straight from c++, you need to have the following line as well because this is where QtFirebaseMessaging does some of its own connections for messages and tokens:
    firebase ->componentComplete();
